engine/lib/views.php

```diff
@@ -1371,60 +1371,105 @@ function elgg_get_form_footer() {
+
+	// $vars passed to label, help and field wrapper views
+	$element_vars = $params;
+
+	// We are adding hashed parameter names to avoid name collisions with input $vars.
+	// To stay BC, we need to remap old key names to new ones
+	$map = [
+		'#label' => 'label',
+		'#help' => 'help',
+		'#class' => 'field_class',
+	];
 
-	$label = elgg_view('elements/forms/label', $vars);
-	if ($input_type == 'checkbox') {
-		$vars['label'] = $label;
-		$vars['label_tag'] = 'div';
-		$label = false;
-	} else {
-		unset($vars['label']);
+	foreach ($map as $new_key => $old_key) {
+		if (isset($params[$new_key])) {
+			// We have a new key definition and we need to map it to keys expected by
+			// label/help views. We don't need to clear these values from $vars,
+			// as they will not be clashing with input view $vars
+			$element_vars[$old_key] = $params[$new_key];
+		} else {
+			// We assume that the value passed with the old key name is meant for
+			// the field wrapper and not the input view
+			unset($input_vars[$old_key]);
+		}
 	}
 
-	$help = elgg_view('elements/forms/help', $vars);
-	unset($vars['help']);
+	$element_vars['class'] = $element_vars['field_class'];
+	$element_vars['label'] = elgg_view('elements/forms/label', $element_vars);
+	$element_vars['help'] = elgg_view('elements/forms/help', $element_vars);
 
-	$required = elgg_extract('required', $vars);
-
-	$field_class = (array) elgg_extract('field_class', $vars, array());
-	unset($vars['field_class']);
+	if ($input_type == 'checkbox') {
+		// Single checkbox input view gets special treatment
+		// We don't want the field label to appear a checkbox without a label
+		$input_vars['label'] = $element_vars['label'];
+		$input_vars['label_tag'] = 'div';
+		unset($element_vars['label']);
+	}
+	
+	$element_vars['input'] = elgg_view("elements/forms/input", $input_vars);
 
-	$input = elgg_view("elements/forms/input", $vars);
+	return elgg_view('elements/forms/field', $element_vars);
+}
 
-	return elgg_view('elements/forms/field', array(
-		'label' => $label,
-		'help' => $help,
-		'required' => $required,
-		'id' => $id,
-		'input' => $input,
-		'class' => $field_class,
-		'input_type' => $input_type,
-	));
+/**
+ * Renders a form field
+ * A shortcut method to rendering inputs from an array of options
+ *
+ * @see elgg_view_input()
+ *
+ * @param array  $params Field parameter and input variables
+ *                       Field vars contain both field and input params.
+ *                       Field parameters are prefixed with a hash (#) to prevent name collisions.
+ *                        - #type: input type
+ *                        - #label: field label
+ *                        - #help: field help text
+ *                        - #class: field class
+ *                       Both #label and #help params accept HTML,
+ *                       and will be printed unescaped within their wrapper element.
+ * @return string
+ */
+function elgg_view_field(array $params = []) {
```

Comment:
Instead of adding another function, why not just rewrite elgg_view_input() and document it to pass params as the first argument (the docs above are fine). But if the 2.2 arguments are given (string, array) then use 2.2 behavior.